express-api/src/services/projects/projectsServices.ts

```diff
@@ -45,17 +110,26 @@ const addProject = async (project: DeepPartial<Project>, propertyIds: ProjectPro
     const newProject = await projectRepo.save(project);
     // After project is saved, add parcel/building relations
     const { parcels, buildings } = propertyIds;
-    if (propertyIds.parcels) await addProjectParcelRelations(newProject, parcels);
-    if (propertyIds.buildings) await addProjectBuildingRelations(newProject, buildings);
+    if (parcels) await addProjectParcelRelations(newProject, parcels);
+    if (buildings) await addProjectBuildingRelations(newProject, buildings);
```

Comment:
should we add checks to see if a project parcel/building already exists on the table with these values?

Reply:
Added some better checks for the following:

- Does that property already have a relation with a project that isn't cancelled or denied?
- Is there already a relation for this property and project? If so, don't try to add again.

Reply:
No longer adds a duplicate entry!
I am noticing that this process will add 2 rows to the project_property table, splitting the parcel and building relations, if this is what we want it to do i am happy with it, but it seems odd to me. 
for example if I run: 
```
const sample_project = await projectServices.getProjectById(6);
const result = await projectServices.addProject(sample_project, {parcels:[6475], buildings:[89]});
```

It results in two entries to the project_property table